Allow to set SL_API_ENDPOINT environment variable

mattcui opened the pull request:
To address [issue 118](https://github.com/maximilien/softlayer-go/issues/118), users can set environment variable SL_API_ENDPOINT to change the softlayer endpoint server.

maximilien commented:
OK @mattcui but what about a simple test that verifies that when `SL_API_ENDPOINT` is set correct value is used in endpoint and when not default. Feel fere to add a `GetApiEndpoint() string` to the `client.go` `Client` interface and to the implementation in `softlayer_client.go` so that way we can test it.

Thanks,

Max

mattcui commented:
@maximilien It's not easy to add it into Client interface, since the api endpoint url needs to be set in [NewSoftLayerClient](https://github.com/maximilien/softlayer-go/blob/master/client/softlayer_client.go#L24), I made some enhancement there, changed it to a public function, there will verify if it's a known api endpoint server, and set to "api.softlayer.com" if it's not set or it's set wrongly. I also added 2 unit test cases. Please review. Thanks.

maximilien commented on client/softlayer_client.go:
```diff
@@ -174,6 +176,24 @@ func (slc *SoftLayerClient) GetSoftLayer_Dns_Domain_ResourceRecord_Service() (so
 
 //Private methods
```
That's not private... capital letter `GetSLApiEndpoint()`

maximilien commented on client/softlayer_client_test.go:
```diff
@@ -187,4 +187,20 @@ var _ = Describe("SoftLayerClient", func() {
 		})
 	})
+
+	Context("#getApiEndpoint, when SL_API_ENDPOINT is set correctly", func() {
```
This should be: `"#GetApiEndpoint, when SL_API_ENDPOINT is set correctly"`

maximilien commented:
@mattcui Merging and I'll address my two comments. Thx
